ext/php8/handlers_curl.c

```diff
@@ -79,54 +76,43 @@ static void dd_ch_duplicate_headers(zval *ch_orig, zval *ch_new) {
     }
 }
 
-static void dd_init_headers_arg(zval *arg, zval *ch) {
-    HashTable *retval;
-    ALLOC_HASHTABLE(retval);
-    HashTable *headers = NULL;
-
-    if (dd_headers) {
-        headers = zend_hash_index_find_ptr(dd_headers, Z_OBJ_HANDLE_P(ch));
-        if (headers) {
-            size_t headers_count = zend_hash_num_elements(headers);
-            zend_hash_init(retval, headers_count, NULL, ZVAL_PTR_DTOR, 0);
-            zend_hash_copy(retval, headers, (copy_ctor_func_t)zval_add_ref);
-        }
+static void dd_inject_distributed_tracing_headers(zval *ch) {
+    zval headers;
+    zend_array *dd_header_array;
+    if (dd_headers && (dd_header_array = zend_hash_index_find_ptr(dd_headers, Z_OBJ_HANDLE_P(ch)))) {
+        ZVAL_ARR(&headers, zend_array_dup(dd_header_array));
+    } else {
+        array_init(&headers);
     }
 
-    if (!headers) {
-        zend_hash_init(retval, 0, NULL, ZVAL_PTR_DTOR, 0);
+    int sampling_priority;
+    if (ddtrace_fetch_prioritySampling_from_root(&sampling_priority)) {
+        add_next_index_str(&headers, zend_strpprintf(0, "x-datadog-sampling-priority: %d", sampling_priority));
+    }
+    if (DDTRACE_G(trace_id)) {
+        add_next_index_str(&headers, zend_strpprintf(0, "x-datadog-trace-id: %" PRIu64, (DDTRACE_G(trace_id))));
+    }
+    if (DDTRACE_G(span_ids_top)) {
+        add_next_index_str(&headers,
+                           zend_strpprintf(0, "x-datadog-parent-id: %" PRIu64, (DDTRACE_G(span_ids_top)->id)));
+    }
```

Comment:
Should we ensure that these are both sent or neither sent?

Reply:
don't know. I don't know either if that makes a big difference too.

Reply:
Only trace id it is worth sending it. At least the two chunks would be visible in our UI, even if disconnected.

Only span id would make no sense, but all the tracers I worked with, when they don't find a trace id they just ignore distributed tracing. So from the functional standpoint there is no harm (actually it is one info more when we try to debug, if span id is present it means that distributed tracing injection was not skipped, but trace id was empty).

In any case (not necessary for this PR unless you have to change other things), I would log an error message in debug mode since this is a misbehavior.